code/datums/components/shrink.dm

```diff
@@ -0,0 +1,43 @@
+/datum/component/shrink
+	var/olddens
+	var/oldopac
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+
+/datum/component/shrink/Initialize(shrink_time)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/atom/parent_atom = parent
+	parent_atom.transform = parent_atom.transform.Scale(0.5,0.5)
+	olddens = parent_atom.density
+	oldopac = parent_atom.opacity
+	parent_atom.density = 0
+	parent_atom.opacity = 0
+	if(isliving(parent_atom))
```

Comment:
because of this if/else, you won't get any message about shrinking if you are a simplemob, which is weird because nonliving things and carbons get a visible message.